src/decoderbufs.c

```diff
@@ -607,9 +608,8 @@ static int tuple_to_tuple_msg(Decoderbufs__DatumMessage **tmsg,
     /* query output function */
     getTypeOutputInfo(attr->atttypid, &typoutput, &typisvarlena);
     if (!isnull) {
-      if (typisvarlena && VARATT_IS_EXTERNAL_ONDISK(origval)) {
-        datum_msg.datum_unchanged = true;
-        datum_msg.datum_case = DECODERBUFS__DATUM_MESSAGE__DATUM_DATUM_UNCHANGED;
+      if (typisvarlena && VARATT_IS_EXTERNAL_ONDISK(tuple)) {
+        set_datum_value(&datum_msg, attr->atttypid, typoutput, heap_tuple_untoast_attr(attr));
```

Comment:
few points on this

* Small refactoring would help - set_datum_value is called in all 3 cases now, only the last param is different, so it can be moved out of it
* datum_unchanged is not used anymore. Is that correct?

Reply:
Yeah, datum_unchanged is not used with this anymore